src/Nox.Types/Types/CultureCode/CultureCode.cs

```diff
@@ -15,14 +29,41 @@ internal override ValidationResult Validate()
     {
         var result = base.Validate();
 
-        if (!CultureCodeRegex().IsMatch(Value))
+        if (!SupportedCultureCodes.Contains(Value))
         {
             result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox CultureCode type with unsupported value '{Value}'."));
         }
 
         return result;
     }
+    
+    /// <summary>
+    /// Creates a new instance of <see cref="CultureCode"/>
+    /// <param name="value">The Culture value type of <see cref="Culture"/> to create the <see cref="CultureCode"/> with</param>
+    /// </summary>
+    /// <returns>A new instance of <see cref="CultureCode"/></returns>
+    public static CultureCode From(Culture value)
+    {
+       
+        var field =  value.GetType().GetField(value.ToString());
+        DisplayNameAttribute? attribute = null;
```

Comment:
suggest to add the computation of the DisplayName/Name to an extension method and keep this metho more simple to read and maintain

Actually I already see that you have it, so use it :)

Reply:
Desired changens have been applied